android-core/src/main/java/com/mparticle/networking/MParticleBaseClientImpl.java

```diff
@@ -27,6 +28,8 @@ public class MParticleBaseClientImpl implements MParticleBaseClient {
     private ConfigManager mConfigManager;
     private BaseNetworkConnection mRequestHandler;
     private SharedPreferences mPreferences;
+
+    private Long currentMPId;
```

Comment:
this is a workaround to avoid injecting the mpid directly? a little bit hard to read/maintain the code, makes me concerned we'll break it in the future.

how does identity modify do it? can we just append the mpid after the fact?